modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -100,7 +104,7 @@ public class Replica {
 
     private final ClockService clockService;
 
-    private final CompletableFuture<Void> waitForActualStateFuture = new CompletableFuture<>();
+    private final AtomicReference<CompletableFuture<Void>> waitForActualStateFuture = new AtomicReference<>();
```

Comment:
Why do we need this change?

Reply:
It is a better way to avoid starting to wait for the actual state process when several messages are sent.
When one message is in process, the other will wait for the future.